core/src/com/unciv/logic/automation/city/ConstructionAutomation.kt

```diff
@@ -348,7 +312,7 @@ class ConstructionAutomation(val cityConstructions: CityConstructions) {
             buildingStats.food *= 3
         }
 
-        if (buildingStats.gold < 0 && civInfo.gold < 0) {
+        if (buildingStats.gold < 0 && civInfo.stats.statsForNextTurn.gold < 10) {
```

Comment:
Stats for next turn is *per turn gold*
You can be in deficit even if you're at -9 gpt
Maybe this can have 2 stages: *1.5 if current deficit and gpt < 0, and another *1.5 if gpt < 10

Reply:
Yes, the reasoning behind this is because buildings generally take multiple turns to build.
If our income is low, then we should pay more attention to the gold production/maintenance cost of the building.
Instead of: If we are temporarily in debt then we should focus on building a gold production building.